### DISCUSSION TITLE
How to hide page title?

### DISCUSSION THREAD

**Discussion by paulocoutinhox** [mkdocs/mkdocs#2431] [Q&A]
Hi,

How to hide page title?

If i remove "# Home" a new title appear with the name of file.

![image](https://user-images.githubusercontent.com/395096/120093173-d5675300-c0ee-11eb-8729-a4cf3e7d5b8e.png)

How i can remove it without hacks?

Thanks.

**Comment by waylan** [comment]
There are four ways to define a page tile as documented [here](https://www.mkdocs.org/user-guide/writing-your-docs/#meta-data). You could use either one of the first two of those options.

> 1. A title defined in the nav configuration setting for a document.
> 2. A title defined in the `title` meta-data key of a document.

In either case, do not include the title in the body of the document. So long as you have defined it in either the nav configuration for the meta-data for the page, the filename will not be used as the title.

Finally, I am moving this to a discussion as it is a support request.

**Reply by tiny-dancer** [reply]
@waylan  i may be missing it but your response seems to cover ways to _define_ a title - how does one _remove_ a title?

e.g.
```
---
hide:
  - title
---
```

**Reply by waylan** [reply]
@tiny-dancer in the original question, the title was part of the body text. By defining the title outside of the body text, then it would not be included on the page. That said, if a theme were to display the title separate from the body text, then you would need to alter the theme to no longer display the title for the page.

Note that this all assumes that the page is still listed in the site Navigation using the title of the page, If instead, you meant to ask how to not include a page in the site navigation, well that is a different question. For that see [Configure Pages and Navigation][1] where is says (last paragraph in that section):

> Any pages not listed in your navigation configuration will still be rendered and included with the built site, however, they will not be linked from the global navigation and will not be included in the `previous` and `next` links. Such pages will be "hidden" unless linked to directly.

[1]: https://www.mkdocs.org/user-guide/writing-your-docs/#configure-pages-and-navigation

**Comment by fellipematos** [comment]
Insert at the beginning of the document.md

```
---
title: NAV TITLE 
---
#
```

**Reply by shenweiyan** [reply]
I have tried it, but it doesn't work.

**Reply by ChristianKniep** [reply]
Add a `#` at the top of your markdown. This will add an empty title. It adds white space at the beginning - so there might be a better way eventually...
EDIT: Found a better way: https://github.com/squidfunk/mkdocs-material/issues/2163#issuecomment-752916358
Just add the `<style>` snippet to your markdown.
```markdown
---
hide:
 - navigation
 - toc
---
<style>
  .md-typeset h1,
  .md-content__button {
    display: none;
  }
</style>
```

**Reply by zillionare** [reply]
this could cause menu failed to be expanded under mobile web page

**Comment by redtide** [comment]
I was about suggesting to override the content block but it seems in the mkdocs theme [the title is part of the page content](https://github.com/mkdocs/mkdocs/blob/master/mkdocs/themes/mkdocs/content.html)?

otherwise in my custom theme I did:

```jinja2
<article>
    {%- if not page.meta.no_title_header %}<h1>{{ page.title }}</h1>{%- endif %}
    {{- page.content }}
</article>
```
So that the title, defined in the frontmatter / page metadata of course, not in the markdown, continues to be used for the html title but hidden in the page.
